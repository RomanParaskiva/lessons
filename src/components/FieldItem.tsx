import React, { useState } from 'react'
import '../styles/fieldItem.css'

interface IFieldItem {
    id?: number | null,
    checked?: boolean | null,
    width?: number
}

const FieldItem = ({ id, checked, width }: IFieldItem) => {
    const [isChecked, setIsChecked] = useState(checked)

    const handleClick = (): void => {
        setIsChecked((prev) => !prev)
    }
    return (
        <div style={{
            width: width + 'px',
            height: width + 'px',
            backgroundColor: isChecked ? 'purple' : '',
            color: isChecked ? 'white' : ''
        }}
            className="fieldItem"
            onClick={handleClick}
        >
            {id}
        </div>
    )
}


export { FieldItem }
export type { IFieldItem }