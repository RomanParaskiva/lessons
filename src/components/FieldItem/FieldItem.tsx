import React, { useState } from 'react'
import './fieldItem.css'

interface IFieldItem {
    id?: number | null,
    checked?: boolean | null,
    width?: number,
}

const FieldItem = ({ id, checked, width }: IFieldItem) => {
    const [isChecked, setIsChecked] = useState(checked)

    const handleClick = () => {
        setIsChecked((prev) => !prev)
        console.log(id)     
    }
    
    return (
        <div style={{
            width: width + 'px',
            height: width + 'px',
            backgroundColor: isChecked ? 'purple' : '',
            color: isChecked ? 'white' : ''
        }}
            className="fieldItem"
            onClick={handleClick}
        >
            {id}
        </div>
    )
}


export { FieldItem }
export type { IFieldItem }