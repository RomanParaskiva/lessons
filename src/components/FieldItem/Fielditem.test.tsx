import { render, fireEvent } from '@testing-library/react'

import { FieldItem } from './FieldItem'

describe('FieldItem', () => {
  test('renders Field component', () => {
    
    const { container } = render(<FieldItem id={1} checked width={50} />);
   
    // fireEvent.click(container, new MouseEvent('click', {
    //   bubbles: true,
    //   cancelable: true,
    // }))
    // expect(container).toBeCalledTimes(1)
    expect(container).toMatchSnapshot()
  });
});