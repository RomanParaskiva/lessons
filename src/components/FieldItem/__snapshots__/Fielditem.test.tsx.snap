// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldItem renders Field component 1`] = `
<div>
  <div
    class="fieldItem"
    id="1"
    style="width: 50px; height: 50px; background-color: purple; color: white;"
  >
    <span>
      1
    </span>
  </div>
</div>
`;
